pallets/nfts/src/lib.rs

```diff
@@ -693,6 +728,23 @@ pub mod pallet {
 		CollectionNotEmpty,
 		/// The witness data should be provided.
 		WitnessRequired,
+		/// The account owns zero items in the collection.
+		NoItemOwned,
+		/// The collection has existing approvals.
+		CollectionApprovalsExist,
+		/// Collection and item approval conflicts.
+		DelegateApprovalConflict,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
+		#[cfg(any(feature = "std", test))]
+		fn integrity_test() {
+			assert!(
+				size_of::<<T as Config<I>>::ItemId>() <= size_of::<u32>(),
```

Comment:
Surely this will fail with an xcm Location? It might need to be a case of checking that itemid != u64/u128 via TypeId